create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/ScopusConverter.java

```diff
@@ -61,9 +65,39 @@ private EntityDescription generateEntityDescription() {
         entityDescription.setReference(generateReference());
         entityDescription.setMainTitle(extractMainTitle());
         entityDescription.setContributors(generateContributors());
+        entityDescription.setTags(generatePlainTextTags());
         return entityDescription;
     }
 
+    private List<String> generatePlainTextTags() {
+        var authorKeywordsTp = extractAuthorKeyWords();
+        return authorKeywordsTp == null
+                   ? null
+                   : authorKeywordsTp.getAuthorKeyword()
+                       .stream()
+                       .map(something -> something.getContent().stream().map(
+                           this::extractContentString).collect(Collectors.joining()))
+                       .collect(Collectors.toList());
+    }
+
+    private String extractContentString(Object content) {
+        if (content instanceof String) {
+            return ((String) content).trim();
+        } else if (content instanceof JAXBElement) {
+            return extractContentString(((JAXBElement<?>) content).getValue());
+        } else if (content instanceof SupTp) {
+            return extractContentString(((SupTp) content).getContent());
+        } else if (content instanceof InfTp) {
+            return extractContentString(((InfTp) content).getContent());
+        } else if (content instanceof ArrayList) {
+            return ((ArrayList<?>) content).stream()
+                .map(this::extractContentString)
+                .collect(Collectors.joining());
+        } else {
+            return content.toString();
+        }
+    }
```

Comment:
So this function is not the most beautiful I have ever seen.
Tried to refactor into something not recursive, but it ended up being spaghetti-code kinda recursive with extra function-calls, but if you look at it as a whole it was still recursive.

last if-statement is just a fail safe. I dont think it's possible in practise that the object is something other than string, jaxbelement, suptp, inftp, or arraylist. Was considering making the last part return empty string instead or throw an exception.

Reply:
Also: this function would extract titles, abstracts etc also as plaintext if so desired.